cli/internal/cmd/apply.go

```diff
@@ -0,0 +1,505 @@
+	checker := license.NewChecker(a.quotaChecker, a.fileHandler)
+	if err := checker.CheckLicense(cmd.Context(), conf.GetProvider(), conf.Provider, cmd.Printf); err != nil {
+		cmd.PrintErrf("License check failed: %v", err)
+	}
+	a.log.Debugf("Checked license")
+
+	// Check if we already have a running Kubernetes cluster
+	// by checking if the Kubernetes admin config file exists
+	// If not, we need to run the init RPC first
+	a.log.Debugf("Checking if %s exists", a.flags.pathPrefixer.PrefixPrintablePath(constants.AdminConfFilename))
+	initRequired := false
+	if _, err := a.fileHandler.Stat(constants.AdminConfFilename); err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf("checking for %s: %w", a.flags.pathPrefixer.PrefixPrintablePath(constants.AdminConfFilename), err)
+		}
+		// Only run init RPC if we are not skipping the init phase
+		// This may break things further down the line
+		// It is the user's responsibility to make sure the cluster is in a valid state
+		initRequired = true && !a.flags.skipPhases.contains(skipInitPhase)
+	}
+	a.log.Debugf("Init RPC required: %t", initRequired)
+
+	// Validate input arguments
+
+	// Validate Kubernetes version as set in the user's config
+	// If we need to run the init RPC, the version has to be valid
+	// Otherwise, we are able to use an outdated version, meaning we skip the K8s upgrade
+	a.log.Debugf("Validating Kubernetes version %s", conf.KubernetesVersion)
+	validVersion, err := versions.NewValidK8sVersion(string(conf.KubernetesVersion), true)
+	if err != nil {
+		a.log.Debugf("Kubernetes version not valid: %s", err)
+		if initRequired {
+			return err
+		}
+		a.log.Debugf("Checking if user wants to continue anyway")
+		if !a.flags.yes {
+			confirmed, err := askToConfirm(cmd,
+				fmt.Sprintf(
+					"WARNING: The Kubernetes patch version %s is not supported. If you continue, Kubernetes upgrades will be skipped. Do you want to continue anyway?",
+					validVersion,
+				),
+			)
+			if err != nil {
+				return fmt.Errorf("asking for confirmation: %w", err)
+			}
+			if !confirmed {
+				return fmt.Errorf("aborted by user")
+			}
+		}
+		a.flags.skipPhases = append(a.flags.skipPhases, skipK8sPhase)
+		a.log.Debugf("Outdated Kubernetes version accepted, Kubernetes upgrade will be skipped")
+	}
+	if versions.IsPreviewK8sVersion(validVersion) {
+		cmd.PrintErrf("Warning: Constellation with Kubernetes %s is still in preview. Use only for evaluation purposes.\n", validVersion)
+	}
+	conf.KubernetesVersion = validVersion
+	a.log.Debugf("Target Kubernetes version set to %s", conf.KubernetesVersion)
+
+	// Validate microservice version (helm versions) in the user's config matches the version of the CLI
+	// This makes sure we catch potential errors early, not just after we already ran Terraform migrations or the init RPC
+	if !a.flags.force {
+		if err := validateCLIandConstellationVersionAreEqual(constants.BinaryVersion(), conf.Image, conf.MicroserviceVersion); err != nil {
```

Comment:
should this behavior be similar to K8s to be consistent? i.e. skip helm when not valid and !initRequired?

Reply:
Also added a similar check for Helm